presto-main/src/main/java/io/prestosql/operator/aggregation/AbstractMinMaxNAggregationFunction.java

```diff
@@ -84,31 +89,36 @@ protected AbstractMinMaxNAggregationFunction(String name, Function<Type, BlockPo
                         AGGREGATE),
                 true,
                 false);
-        requireNonNull(typeToComparison);
-        this.typeToComparator = typeToComparison;
+        this.min = min;
     }
 
     @Override
     public List<TypeSignature> getIntermediateTypes(FunctionBinding functionBinding)
     {
         Type type = functionBinding.getTypeVariable("E");
-        return ImmutableList.of(new MinMaxNStateSerializer(typeToComparator.apply(type), type).getSerializedType().getTypeSignature());
+        return ImmutableList.of(RowType.anonymous(ImmutableList.of(BIGINT, type)).getTypeSignature());
```

Comment:
Rather than hard coding this directly here, it feels a bit better to pull it out directly from the MinMaxNStateSerializer which should be the one defining this intermediate type. Maybe add a static method to MinMaxNStateSerializer to fetch this?

Reply:
Ya this code is a bit of a mess.  I clean this up as part of the function refactoring code.